File: src/services/orderService.js
import request from './request';

export const OrderService = {
    // 创建订单
    createOrder: async (userId, shippingAddress) => {
        return request(`/orders/create?userId=${userId}&shippingAddress=${encodeURIComponent(shippingAddress)}`, {
            method: 'POST'
        });
    },

    // 获取用户的订单列表
    getUserOrders: async (userId) => {
        return request(`/orders/user/${userId}`);
    },

    // 获取订单详情
    getOrderById: async (orderId) => {
        return request(`/orders/${orderId}`);
    },

    // 更新订单状态
    updateOrderStatus: async (orderId, status) => {
        return request(`/orders/${orderId}/status`, {
            method: 'PUT',
            body: JSON.stringify({ status }),
        });
    },
};

File: src/main/java/me/cozease/bookstorebackend/service/OrderService.java
package me.cozease.bookstorebackend.service;

import me.cozease.bookstorebackend.entity.*;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * 订单服务类
 */
@Transactional
public interface OrderService {
    /**
     * 创建订单
     */
    Order createOrder(Long userId, String shippingAddress);

    /**
     * 获取用户的所有订单
     */
    List<Order> getUserOrders(Long userId);

    /**
     * 获取订单详情
     */
    Order getOrderById(Long orderId);

    /**
     * 更新订单状态
     */
    Order updateOrderStatus(Long orderId, String status);
}
